crates/apollo-compiler/src/ast/impls.rs

```diff
@@ -71,6 +81,17 @@ impl Document {
         errors.into_result_with(document)
     }
 
+    /// Build and validate an executable document from this AST, with the given schema
+    pub fn to_executable_validate(
+        &self,
+        schema: &Valid<Schema>,
+    ) -> Result<Valid<ExecutableDocument>, WithErrors<ExecutableDocument>> {
+        let mut errors = DiagnosticList::new(self.sources.clone());
+        let document = self.to_executable_inner(schema, &mut errors);
+        crate::executable::validation::validate_standalone_executable(&mut errors, &document);
```

Comment:
this should use full operation validation right?
```suggestion
        crate::executable::validation::validate_executable_document(&mut errors, schema, &document);
```

Reply:
Good catch!